tests/testsuite/script.rs

```diff
@@ -1024,7 +1024,7 @@ fn cmd_metadata_with_embedded() {
                 "default_run": null,
                 "name": "script",
                 "version": "0.0.0",
-                "id": "script[..]",
+                "id": "file:[..]foo#script@0.0.0",
```

Comment:
Not a blocker for this PR but for #12207 is that we need to make sure cargo script PackageIdSpec's are unambiguous.  They currently point to the directory, rather than the script.